doc/examples/segmentation/plot_ncut.py

```diff
@@ -20,12 +20,26 @@
 
 img = data.coffee()
 
-labels1 = segmentation.slic(img, compactness=30, n_segments=400)
-out1 = color.label2rgb(labels1, img, kind='avg')
+super_pixels = segmentation.slic(img, compactness=30, n_segments=400)
+out1 = color.label2rgb(super_pixels, img, kind='avg')
 
-g = graph.rag_mean_color(img, labels1, mode='similarity')
-labels2 = graph.cut_normalized(labels1, g)
-out2 = color.label2rgb(labels2, img, kind='avg')
+g = graph.rag_mean_color(img, super_pixels, mode='similarity')
+
+# Two ways to apply Normalized Cuts:
+
+# Getting the value for a single threshold
+labels1 = graph.cut_normalized(super_pixels, g, thresh=1e-3)
+
+# Or using a generator to try several
+label_gen = graph.cut_normalized_gen(super_pixels, g, thresh=1e-4)
+labels2 = next(label_gen)
+labels2 = label_gen.send(1e-3)
+
+# Either method yields the same labels, barring an unlucky random seed
```

Comment:
Let's fix the seed for this example then.

Reply:
Unfortunately doing that isn't as easy as calling np.random.seed(0). The random initialization that seems to be causing the problem is the initial residual vector for the Lanczos algorithm. Reading through the source code for eigsh, it looks like that initialization gets performed by ARPACK in dsaupd.f. Do you know of any way to set that seed? 

If not, the best option I can think of is to generate a random vector in _ncut_relabel and use that to set eigsh's v0 argument. 

Should eigsh be modified to set a random initial residual vector itself? I probably wouldn't have realized np.random.seed wasn't fixing the output if I hadn't already been looking into the stability issues. 

As a side note, ARPACK's random initial vector seems to be reproducible - when I get a bad initial vector, it's bad every time I run the code. Other factors seem to affect the choice of initial vector though, since the stability test and gallery example assert statement sometimes break a Travis CI config and sometimes don't.

Reply:
Just to back up the allegation that it's random initialization in ARPACK. When using commit 7168f5f, the labels produced by the cut_normalized_gen and cut_normalized in the gallery example are not the same for me. I was able to make them the same by calling cut_normalized three times in a row. This worked even when setting the numpy seed between each call.

Reply:
If you know the range within which the `v0` is that they generate, then we can certainly set it ourselves.

I suspect SciPy will reject a patch to have v0 be generated by NumPy simply to use the NumPy random seed.  But, it may be worth posting the issue on the mailing list and ask about it.  I can also do that, if you are not subscribed.

Our other option is simply not to compare the results above---i.e., show the API, but don't try to suggest that for different threshold we would get the same results.  That seems like an odd thing to expect anyway?

Reply:
I don't see the v0 range mentioned in the docs for eigsh or dsaupd.f, but Wikipedia says the algorithm uses a norm 1 vector so that would probably work. 

I'm not subscribed to mailing list, so it might be better if you send something out. 

Not sure if I understood. I am comparing two separate results for thresh=1e-3 in the gallery example. For cut_normalized_gen I get results for 1e-4 first then overwrite them with results for 1e-3.